kivy/effects/scroll.py

```diff
@@ -99,11 +99,13 @@ def on_value(self, *args):
         if scroll_min > scroll_max:
             scroll_min, scroll_max = scroll_max, scroll_min
         if self.value < scroll_min:
-            self.overscroll = self.value - scroll_min
-            self.reset(scroll_min)
+            if self.value > scroll_max:
+                self.overscroll = self.value - scroll_min
+                self.reset(scroll_min)
         elif self.value > scroll_max:
-            self.overscroll = self.value - scroll_max
-            self.reset(scroll_max)
+            if self.value < scroll_max:
```

Comment:
How `self.value` could be less than `scroll_max` here, is a typo?

Reply:
scroll_max is always 0 in ScrollEffect and self.value decreases along scroll from top to bottom

Reply:
Considered that to reach line `106` the condition at line `105` is `True`, how can be `self.value < scroll_max` if it was `self.value > scroll_max` ?

Reply:
I understand what you're saying, it doesn't really make sense, it's so that it never resets when it's in this condition, as it has no effect

Reply:
if it resets, the behavior will not be as expected.

Reply:
I agree with @misl6 .

```python
elif A > B:
    if A < B:
```

The if statement looks never be True.

Reply:
I was wrong guys, sorry!! I encountered the ScrollEffect problem.
I had to change the ScrollEffect and the DampedScrollEffect
I will send the corrections

Reply:
I don't fully understand the `kivy/effects/*` (I'm currently trying to), but I think the PR is appropriate. :+1: